eskip/string.go

```diff
@@ -80,7 +80,85 @@ func argsString(args []interface{}) string {
 	return strings.Join(sargs, ", ")
 }
 
-func (r *Route) predicateString(sortPredicates bool) string {
+func (r *Route) predicateStringSorted() string {
+	var predicates []string
+
+	cr := r.Copy()
```

Comment:
Copy transform legacy predicates into r.Predicates.

TBH I am not sure it is best to have two methods `predicateStringSorted` and `predicateString` - I would try to apply sorting like you proposed previously in place where needed inside the if block...

Reply:
then we will need to do this (copy and sort) for all maps/slices (HostRegex, PathRegex, Headers, HeadersRegex and Predicates)

> Copy transforms legacy predicates into r.Predicates

According to this it just make a deep copy, I'm I missing something?  https://github.com/zalando/skipper/blob/master/eskip/eskip.go#L371-L419

Reply:
After some discussions we agreed it would be better to have one function so it's easier to maintain and more error prune.

Another idea was to sort the string after joining, but we will leave this for another day.